src/bootstrap/lib.rs

```diff
@@ -219,6 +222,15 @@ impl Build {
         sanity::check(self);
         self.verbose("collecting channel variables");
         channel::collect(self);
+        // If local-rust is the same as the current version, then force a local-rebuild
+        let local_version = String::from_utf8(
+            Command::new(&self.rustc)
+                .arg("--version").output()
+                .expect("failed to execute rustc --version").stdout).unwrap();
```

Comment:
I believe there's a [`build_helper::output` function](https://github.com/rust-lang/rust/blob/fe5c439f695fd72b7be0d0f28ee8397b3256630c/src/build_helper/lib.rs#L61) which should help with running a command and getting stdout as a string